### PR TITLE
Add refresh parameter config option

### LINKED ISSUE #2581: Replace refresh=true with refresh=wait_for in Fleet Server
While calling Elasticsearch APIs, fleet server uses a refresh flag value that default to `true` this should be change to `wait_for`.

- [ ] This change should be testing against a stateful Elasticsearch first to see if whether or not we want to apply it on all deployment types.

### PR DESCRIPTION
**Pull request by michel-laterman**
## What is the problem this PR solves?

Fleet-server should not force ES to refresh with all of it's requests.

## How does this PR solve the problem?

Add a config option that allows us to define the refresh parameter value instead of using `refresh=true` for all requests. 
Parameter is used for bulk operations and creating chunks while uploading a file.
Parameter is not used (and `true` is used) for [mget search](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-multi-get.html#docs-multi-get-api-query-params) requests and the [update by query](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update-by-query.html#docs-update-by-query-api-query-params) requests as refresh is still defined as a bool param.

## Design Checklist

- [x] I have ensured my design is stateless and will work when multiple fleet-server instances are behind a load balancer.
- [ ] I have or intend to scale test my changes, ensuring it will work reliably with 100K+ agents connected.
- ~~I have included fail safe mechanisms to limit the load on fleet-server: rate limiting, circuit breakers, caching, load shedding, etc.~~

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- ~~I have made corresponding changes to the documentation~~
- [x] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/fleet-server#changelog)

## Related issues

- Closes #2581

### CODE REVIEW

**Comment by joshdover** [comment]
Seems like we'll need to understand which refresh changes cause problems. Probably worth running the integration tests with fewer changes and see which one starts breaking things.

I would avoid us spending too much time on this and instead try utilizing the `fast_refresh` setting if needed.

File: changelog/fragments/1689704810-add-refresh-param-config.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+description: Add a parameter that allows for the change of the refresh param from "true" to "wait_for" that effects file upload and bulk creation calls.
+
+# Affected component; a word indicating the component this changeset affects.
+component:
+
+# PR URL; optional; the PR number that added the changeset.
+# If not present is automatically filled by the tooling finding the PR where this changelog fragment has been added.
+# NOTE: the tooling supports backports, so it's able to fill the original PR number instead of the backport PR number.
+# Please provide it if you are adding a fragment for a different PR.
+#pr: https://github.com/owner/repo/1234
```
Comment (juliaElastic) [inline changelog/fragments/1689704810-add-refresh-param-config.yaml:28]:
```suggestion
pr: 2799
```

**Comment by joshdover** [comment]
@michel-laterman should we close this in favor of https://github.com/elastic/elasticsearch-serverless/pull/600?

**Comment by michel-laterman** [comment]
Yeah, I'll close the extra PRs

**Pull request closed without merging**